File: src/KubeOps.Templates/Templates/Operator.CSharp/Webhooks/DemoMutator.cs
using KubeOps.Operator.Webhooks;
using GeneratedOperatorProject.Entities;

namespace GeneratedOperatorProject.Webhooks;

public class DemoMutator : IMutationWebhook<V1DemoEntity>
{
    public AdmissionOperations Operations => AdmissionOperations.Create;

    public MutationResult Create(V1DemoEntity newEntity, bool dryRun)
    {
        newEntity.Spec.Username = "not foobar";
        return MutationResult.Modified(newEntity);
    }
}

File: _old/src/KubeOps.Templates/Templates/Operator.FSharp/Webhooks/DemoMutator.fs
namespace GeneratedOperatorProject.Webhooks

open GeneratedOperatorProject.Entities
open KubeOps.Operator.Webhooks
open Microsoft.AspNetCore.Http

type DemoMutator() =
    interface IMutationWebhook<V1DemoEntity> with
        member this.Operations = AdmissionOperations.Create

        member this.Create(newEntity, _) =
            newEntity.Spec.Username <- "not foobar"
            MutationResult.Modified(newEntity)
